components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt

```diff
@@ -816,6 +815,25 @@ class PromptFeature private constructor(
         dialog.feature = this
 
         if (canShowThisPrompt(promptRequest)) {
+            // If the ChoiceDialogFragment's choices data were updated,
+            // we need to dismiss the previous dialog
+            activePrompt?.get()?.let { promptDialog ->
+                // ChoiceDialogFragment could update their choices data,
+                // and we need to dismiss the previous UI dialog,
+                // without consuming the engine callbacks, and allow to create a new dialog with the
+                // updated data.
+                if (promptDialog is ChoiceDialogFragment &&
+                    !session.content.promptRequests
+                        .map { prompt -> prompt.uid }
+                        .contains(promptDialog.promptRequestUID)
+                ) {
```

Comment:
nit: Maybe `!session.content.promptRequests.any { it.uid == promptDialog.promptRequestUID }`